### PR TITLE
ENT-3522: Unset learners language so that default_language from enterprise customer may take effect.

### PR DESCRIPTION
**Pull request by saleem-latif**
**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [x] `make static` has been run to update webpack bundling if any static content was updated
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [x] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

**Review by sameenfatima78 — COMMENTED** [review]
Aren't we adding any more test cases to test this entire functionality?

File: enterprise/admin/__init__.py [hunk 1]
```diff
@@ -193,6 +198,18 @@ class EnterpriseCustomerAdmin(DjangoObjectActions, SimpleHistoryAdmin):
...
         model = EnterpriseCustomer
 
+    def save_model(self, request, obj, form, change):
+        """
+        Check if `default_language` has changed or not, if it is changed then update language preference of all the
+        learners belonging to this enterprise customer.
+        """
+        super(EnterpriseCustomerAdmin, self).save_model(request, obj, form, change)
+
+        if 'default_language' in form.changed_data and obj.default_language:
```
[the remark this answers is not in the document]
Comment (saleem-latif) [reply enterprise/admin/__init__.py:208]: No, actually we want to perform this action after the model has successfully saved.

**Comment by saleem-latif** [comment]
@sameenfatima78 I have updated the existing test cases that I added in https://github.com/edx/edx-enterprise/pull/1002 to cater to these changes.

**Review by sameenfatima78 — APPROVED** [review]
LGTM!

File: enterprise/utils.py [hunk 2]
```diff
@@ -1227,3 +1238,34 @@ def delete_tableau_user_by_id(tableau_user_id):
             '[TABLEAU USER SYNC] Could not delete the tableau user %s.',
             tableau_user_id,
         )
+
+
+def unset_language_of_all_enterprise_learners(enterprise_customer):
```
Comment (irfanuddinahmad) [inline enterprise/utils.py:1243]: Is this part of the solution proposed in the JIRA ticket ENT-3522?

**Review by irfanuddinahmad — COMMENTED** [review]
The solution in this PR is a reactive approach to fix the original issue i.e. English as the default language for all users!

**Review by irfanuddinahmad — APPROVED** [review]

File: enterprise/utils.py (hunk 2 above)
Comment (saleem-latif) [reply enterprise/utils.py:1243]: No, during testing I found, I had to do these too in order for the solution to work correctly.

**Pull request merged**